js/locale/locale.js

```diff
@@ -53,14 +54,15 @@ export class LocalePlugin extends RipeCommonsPlugin {
         return key in keys;
     }
 
-    toLocale(key, defaultValue, locale, fallback = true) {
-        return this.getLocaleValue(key, defaultValue, locale, fallback);
+    toLocale(key, defaultValue, locale, fallback = true, localeMap) {
+        return this.getLocaleValue(key, defaultValue, locale, fallback, localeMap);
     }
 
     async setLocale(locale, coerce = true) {
         if (coerce) locale = this._coerceLocale(locale);
         await this.owner.trigger("pre_set_locale", locale);
         this.locale = locale;
+        this.localeMap = await this._loadLocales();

```

Comment:
Why is this needed?

Reply:
This is needed to assign the new `localeMap` when a different locale is set.
With this, `localeMap` change will trigger the computed `locale()` which fixes the bug

Reply:
I still don't get why, because this function changes `this.locale = locale` and we're already taking care of that in:

```
localePlugin.owner.bind("post_set_locale", locale => (this.currentLocale = locale));
```

Let's go over this on Discord. Please ping me

Reply:
I still think we need this line after your suggestion https://github.com/ripe-tech/ripe-commons-pluginus/blob/master/vue/plugins/locale.js#L37
post_set_locale is only setting the currentLocale with old `localeMap`.
Only after `await _loadLocales` `localeMap` changes and is able to trigger it.

Closing and opening modal works without this line but it also works in master.
I believe this fix could potentially solve the need to `forceUpdate` in reference and pickers

Reply:
Before this line, when did the `localeMap` get updated?

Reply:
in `loadModelLocales` https://github.com/ripe-tech/ripe-commons-pluginus/blob/master/js/locale/model-locale-loader.js#L24

I am going to refactor this solution as some things have changed ever since this was proposed and it's not the best solution anymore as part of the fix was done outside of this PR but not all of it.

Reply:
Much better now!